### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by sirajudheenam**
1. rubycop suggestions applied.
2. refactor code
3. Add Cloud Watch Client
4. MultiCloud Account activation fixed
5. Sandbox detection

### CODE REVIEW

**Comment by sirajudheenam** [comment]
Applied prettier format to the code.

File: plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  .bs-callout.bs-callout-info.bs-callout-emphasize
+    -# Add edgecase of user terminating AWS account then trying to activate again
+    - if nebula_details.class == Hash && nebula_details.key?('error')
+      %h3 ERROR:
+      = nebula_details['error']
+      %h4 Contact Cronus Support by creating a ticket
+      %p
+        Please refer our documetation on our support procedures
+        = link_to "here", "https://documentation.global.cloud.sap/docs/customer/services/email-service/incident-management/"
```
Comment (ArtieReus) [inline plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml:46]: url doc method

File: plugins/email_service/app/views/email_service/shared/role_warning.html.haml [hunk 2]
```diff
@@ -13,7 +13,31 @@
...
+        Two new roles,
+        %code
+          email_admin
+        and
+        <code>email_user</code>, are available to manage the email service.
+        From now on you will need one of the role to be able to access email service.
+        You can assign these roles to your user or the group you belong to.
+        If you do not have permissions, ask your team member who has permissions to assign relevant role to you.
+        %br
+        = link_to "Read more on role from Converged Cloud documentation", "https://documentation.global.cloud.sap/docs/customer/access-management/identity-service/ccroles/", target: :_blank
```
Comment (ArtieReus) [inline plugins/email_service/app/views/email_service/shared/role_warning.html.haml:29]: doc url method

File: plugins/email_service/app/views/email_service/statistics/_table_domain_statistics_report.html.haml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      %table.table.table-striped
+        %tr
+          %td
+            %strong report
+          %td 
+            =@domain_statistics_report
+    - else 
+      domain statistics report is not generated.
+
+-# resp.overall_volume.domain_isp_placements #=> Array
```
Comment (ArtieReus) [inline plugins/email_service/app/views/email_service/statistics/_table_domain_statistics_report.html.haml:18]: do you need this?

File: plugins/email_service/app/views/layouts/email_service/application.html.erb [hunk 4]
```diff
@@ -5,7 +5,7 @@
   <%= csrf_meta_tags %>
```
Comment (ArtieReus) [inline plugins/email_service/app/views/layouts/email_service/application.html.erb:5]: Why do you need an own application template?

**Review by ArtieReus — CHANGES_REQUESTED** [review]

File: plugins/email_service/app/controllers/email_service/settings_controller.rb [hunk 5]
```diff
@@ -1,36 +1,30 @@
...
-        flash[:error] = error
-        check_ec2_creds_cronus_status
-      end
-    rescue Elektron::Errors::ApiResponse => e
-      error =
-        "#{I18n.t("email_service.errors.cronus_account_details_list")} : #{e.message}"
-      Rails.logger.error error
-      flash[:error] = error
-    rescue Exception => e
+      # unless ec2_creds&.access && ec2_creds&.secret
```
[the remark this answers is not in the document]
Comment (sirajudheenam) [reply plugins/email_service/app/controllers/email_service/settings_controller.rb]: removed.

File: plugins/email_service/app/views/email_service/shared/_nav.html.haml [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
+  %li{class: ('active' if pane == 'emails'), role: "presentation"}= link_to 'Send Email', plugin('email_service').emails_path, aria: {controls:"emails-pane"}, role: "tab"
+  %li{class: ('active' if pane == 'verified-emails'), role: "presentation"}= link_to 'Verify Email', plugin('email_service').email_verifications_path, aria: {controls:"verified-emails-pane"}, role: "tab"
   %li{class: ('active' if pane == 'verified-domains'), role: "presentation"}= link_to 'Verify Domain', plugin('email_service').domain_verifications_path, aria: {controls:"verified-domains-pane"}, role: "tab"
   %li{class: ('active' if pane == 'templates'), role: "presentation"}= link_to 'Templates', plugin('email_service').templates_path, aria: {controls:"templates-pane"}, role: "tab"
   -# %li{class: ('active' if pane == 'custom-templates'), role: "presentation"}= link_to 'Custom Templates', plugin('email_service').custom_verification_email_templates_path, aria: {controls:"custom-templates-pane"}, role: "tab"
```
[the remark this answers is not in the document]
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/shared/_nav.html.haml:6]: still need this as I have added these feature and testing them. I will enable soon.

File: plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          = link_to "here", "https://documentation.global.cloud.sap/docs/customer/services/email-service/incident-management/"
+
+      - elsif current_user.is_allowed?('email_service:multicloud_account_create')
+        %h4 Cronus is not enabled
+        = link_to "Enable Cronus (email service)", plugin('email_service').new_multicloud_account_path, data: { modal: true }, class: 'btn btn-primary'
+        %br
+        %p
+          Enable cronus by executing this command <code>cronuscli nebula activate --account-env PROD --security-officer EMAIL-ADDRESS OR C/D/I NUMBER --mail-type TRANSACTIONAL</code> on webCLI.
+          Please refer our documetation on this
+          = link_to "here", "https://documentation.global.cloud.sap/docs/customer/services/email-service/quick-start-sending-emails/#step-2-activate-email-service"
```
[the remark this answers is not in the document]
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml]: sap doc URL updated.

File: plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml (hunk 1 above)
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml]: sap doc URL updated.

File: plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    .bs-callout.bs-callout-info.bs-callout-emphasize
+      -# Add edgecase of user terminating AWS account then trying to activate again
+      - if nebula_details.key?('error')
+        %h3 ERROR:
+        = nebula_details['error']
+        %h4 Contact Cronus Support by creating a ticket
+        %p
+          Please refer our documetation on our support procedures
+          = link_to "here", "https://documentation.global.cloud.sap/docs/customer/services/email-service/incident-management/"
```
[the remark this answers is not in the document]
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/shared/cronus_activation_warning.html.haml]: sap doc URL updated.

File: plugins/email_service/app/views/email_service/settings/_deliverability_dashboard_options.html.haml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+- if @deliverability_dashboard_options
+  %table.table.table-striped
+    %tr
+      %td dashboard enabled
+      %td
+        = @deliverability_dashboard_options[:dashboard_enabled]
+    -# - @deliverability_dashboard_options[:dashboard_enabled]
```
[the remark this answers is not in the document]
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/settings/_deliverability_dashboard_options.html.haml]: removed.

File: plugins/email_service/app/views/email_service/shared/role_warning.html.haml (hunk 2 above)
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/shared/role_warning.html.haml]: sap doc URL updated.

File: plugins/email_service/app/views/email_service/statistics/_table_domain_statistics_report.html.haml (hunk 3 above)
Comment (sirajudheenam) [reply plugins/email_service/app/views/email_service/statistics/_table_domain_statistics_report.html.haml:18]: removed.

File: plugins/email_service/app/views/layouts/email_service/application.html.erb (hunk 4 above)
Comment (sirajudheenam) [reply plugins/email_service/app/views/layouts/email_service/application.html.erb:5]: it is still needed to do pre checks while landing on the plugin.

**Review by ArtieReus — APPROVED** [review]
lgtm

**Pull request merged**